src/store/mutations.ts

```diff
@@ -185,16 +185,19 @@ class Mutations {
     }
   }
   /**
-   * Delete the step of index `index` in pipeline.
+   * Delete selected steps by indexes in pipeline.
    */
   @resetPagination
-  deleteStep(state: VQBState, { index }: { index: number }) {
+  deleteSteps(state: VQBState, { indexes }: { indexes: number[] }) {
     const pipeline = currentPipeline(state);
-    if (!pipeline) {
+    if (state.currentPipelineName === undefined || pipeline === undefined) {
       return;
     }
-    pipeline.splice(index, 1);
-    state.selectedStepIndex = index - 1;
+    const pipelineWithDeletedSteps = pipeline.filter(
+      (_: PipelineStep, index: number) => indexes.indexOf(index) === -1,
+    );
+    state.pipelines[state.currentPipelineName] = pipelineWithDeletedSteps;
+    state.selectedStepIndex = pipelineWithDeletedSteps.length - 1;
```

Comment:
Mainly update, we may not want to keep the "next selected step" with multiple selected steps ?

Reply:
@austil , en fait précédemment on séléctionnait la step directement dans le pipeline après celle supprimée (ça pouvait être en plein millieu du coup). Maintenant on selectionne forcément la dernière step du pipeline. C'était pour prévenir et demander si ce comportement était ok pour nous, ou si on veux un comportement plus proche de l'ancien (Plutot UX pas loquant pour merger, je verrais ça avec Maxime :wink:)

Reply:
J'ai fais un point avec David on reste sur le comportement actuel pour le moment